api/src/main/java/org/openmrs/module/queue/api/impl/QueueEntryServiceImpl.java

```diff
@@ -225,4 +252,25 @@ private void endQueueEntry(@NotNull QueueEntry queueEntry) {
 		queueEntry.setEndedAt(new Date());
 		dao.createOrUpdate(queueEntry);
 	}
+	
+	@Override
+	@Transactional(readOnly = true)
+	public QueueEntry getPreviousQueueEntry(@NotNull QueueEntry queueEntry) {
+		Queue queueComingFrom = queueEntry.getQueueComingFrom();
+		QueueEntrySearchCriteria criteria = new QueueEntrySearchCriteria();
+		criteria.setPatient(queueEntry.getPatient());
+		criteria.setVisit(queueEntry.getVisit());
+		criteria.setEndedOn(queueEntry.getStartedAt());
+		criteria.setQueues(queueComingFrom == null ? Arrays.asList() : Arrays.asList(queueComingFrom));
```

Comment:
What exactly do you want to happen here?  If you set this to an empty list, then the DAO will only return results in which the queue property of QueueEntry is null.  Since this is not (currently) nullable, this will effectively mean that no results are returned.  If that is _actually_ what we want, then we should just do a check earlier in the method and return null without bothering to hit the DB.  If it is _not_ what we want, and we want to get all queue entries that match regardless of queue (and if there is only one then use it), then we need the if statement to be around this entire critieria inclusion call.